### PR TITLE
Release v0.25 - She'll be Comin' Round the Mountain

### PR DESCRIPTION
**Pull request by ryangribble**
This release has been long overdue 😭

Plenty of goodies in here including the dotnetcore port (netstandard 1.1) and moving to CAKE.Frosting as a build system... several bug fixes and several new API's implemented including Projects, Pull Request Reviews, Pull Request Review Requests and Organization Outside Collaborators!

Im also tweaking the release notes format a bit, and have been working on [a utility](https://github.com/ryangribble/Octokit.ReleaseNotes) to generate them, based on `release_notes` comments on the merged PR's.  Everything you see in ReleaseNotes.md was generated by this utility this time around, so any wording tweaks need to be done "at the source" and then I can regenerate the release notes.  Ultimately I'd like to do some more work so that creation of draft releases and generating release notes etc can be done in CAKE scripts and run by AppVeyor as appropriate.

All integration tests are passing so I'm pretty ready to 🚢 - will aim for tomorrow night.  I just need to go through things and see if there are any breaking changes that need to be called out.

- [x] initial release notes
- [x] version bump (need to see how AppVeyor and GitVersion will deal with this... I THINK once I tag master after mergint this PR, the resultant build should get the correct version!)
- [x] run `build FormatCode` to tidy up whitespace/formatting
- [x] integration tests all pass
- [x] tweak release notes to call out any breaking changes/special instructions
- [x] :thumbsup: from @shiftkey or @Haacked
- [x] :thumbsup: from key contributors @mderriey @alfhenrik  @M-Zuber etc
- [ ] :shipit:

### CODE REVIEW

**Comment by M-Zuber** [comment]
👍 from me, thank you for working so hard on this release!

**Comment by mderriey** [comment]
I had a look, and it all looks great; those release notes look :gem:!

I had another look at the [PR that brought the GitVersion config file](https://github.com/octokit/octokit.net/pull/1555) to refresh the memory, and it makes sense that `master` has a prerelease tag since we use GitHubFlow. Tagging `master` after merging this PR should definitely produce `0.25.0`.

Thanks for driving that release, great effort!

giphy

**Comment by shiftkey** [comment]
Release notes look good, and thanks to everyone who contributed to this - especially @ryangribble and @mderriey for the .NET Core work!

**Comment by ryangribble** [comment]
Can someone have a final check of the Advisories and Breaking Changes section I added to the release notes?

**Comment by mderriey** [comment]
It all reads well to me.

File: ReleaseNotes.md [hunk 1]
```diff
@@ -1,3 +1,78 @@
+### New in 0.25.0 (released 23/8/2017)
+
+## Advisories and Breaking Changes
+
+- Octokit.net has been ported to dotnetcore :tada: providing libraries targetting `netstandard1.1` and `net45` frameworks
+
+- `Enum` fields in Octokit response classes are now wrapped in an `StringEnum<TEnum>` helper class, to provide more robustness in dealing with unknown API values for these fields.  Whilst the changes are backwards compatible, please consult the guidance on [working with Enums](https://github.com/octokit/octokit.net/blob/master/docs/working-with-enums.md) for more information
+
+- `IncludeAdmins` field is no longer present in `BranchProtectionRequiredStatusChecks` and `BranchProtectionRequiredStatusChecksUpdate` classes, instead use the new `EnforceAdmins` field on `BranchProtectionSettingsUpdate` or the [new explicit methods](https://github.com/octokit/octokit.net/blob/master/Octokit/Clients/IRepositoryBranchesClient.cs#L304-L365) for configuring Admin Enforcement on protected branches.  This was an [upstream API breaking change](https://developer.github.com/changes/2017-05-02-adoption-of-admin-enforced/) so couldn't follow our normal deprecation schedule
```
Comment (M-Zuber) [inline ReleaseNotes.md]: > so couldn't follow our normal deprecation schedule

so **we** couldn't

**Comment by ryangribble** [comment]
Travis is having capacity issues with OSX builds, it's been queued for serveral hours.  Given it's just doc updates since the last succesful builds Im going to hit the button on this release!

Everyone stand back...

**Pull request merged**